docs/src/reference/asciidoc/core/kerberos.adoc

```diff
@@ -0,0 +1,484 @@
+https://www.elastic.co/guide/en/elastic-stack-overview/current/kerberos-realm.html[Elastic Stack documentation].
+
+Additionally, you will need to https://www.elastic.co/guide/en/elasticsearch/reference/current/security-settings.html[
+configure the API Key Realm] in {es}. Hadoop and other distributed data processing frameworks only authenticate with
+{krb} in the process that launches a job. Once a job has been launched, the worker processes are often cut off from
+the original {krb} credentials and need some other form of authentication. Hadoop services often provide mechanisms for
+obtaining _Delegation Tokens_ during job submission. These tokens are then distributed to worker processes which use
+the tokens to authenticate on behalf of the user running the job. {ehtm} obtains API Keys in order to provide tokens
+for worker processes to authenticate with.
+
+[[kerberos-settings-eshadoop]]
+==== Connector Settings
+
+The following settings are used to configure {eh} to use {krb} authentication:
+
+`es.security.authentication` (default `simple`, or `basic` if `es.net.http.auth.user` is set)::
+Required. Similar to most Hadoop integrations, this property signals which method to use in order to authenticate with
+{es}. By default, the value is `simple`, unless `es.net.http.auth.user` is set, in which case it will default to
+`basic`. The available options for this setting are `simple` for no authentication, `basic` for basic http
+authentication, `pki` if relying on certificates, and `kerberos` if {krb} authentication over SPNEGO should be used.
+
+`es.net.spnego.auth.elasticsearch.principal` (default none)::
+Required if `es.security.authentication` is set to `kerberos`. Details the name of the service principal that the
+Elasticsearch server is running as. This will usually be of the form `HTTP/node.address@REALM`. Since {es} is
+distributed and should be using a service principal per node, you can use the `_HOST` pattern
+(like so `HTTP/_HOST@REALM`) to have {eh} substitute the address of the node it is communicating with at runtime. Note
+that {eh} will attempt to reverse resolve node IP addresses to hostnames in order to perform this substitution.
+
+`es.net.spnego.auth.mutual` (default false)::
+Optional. The SPNEGO mechanism assumes that authentication may take multiple back and forth request-response cycles for
+a request to be fully accepted by the server. When a request is finally accepted by the server, the response contains a
+payload that can be verified to ensure that the server is the principal they say they are. Setting this to `true`
+instructs {eh} to perform this mutual authentication, and to fail the response if it detects invalid credentials from
+the server.
+
+[[kerberos-hadoop]]
+=== {krb} on Hadoop
+
+[[kerberos-hadoop-requirements]]
+==== Requirements
+
+Before using {krb} authentication to {es}, Hadoop Security must be enabled.
```

Comment:
`Hadoop Security must be enabled` >> `Hadoop Security with Kerberos must be enabled`